deepdives/build-process/README.md

```diff
@@ -0,0 +1,66 @@
+# Deep Dive: The Build Process
+
+## Why?
+As you start to create more and more ambitious apps, the time you invest into a build process which automates work for you, cleans up your code, and gives you access to more features becomes more and more worth it. These tools can be intimidating to learn at first, but with a bit of pre work they can make your life a ton easier.
+
+## The Parts of the Process
+
+### Compilers
+You've already been exposed to these with SASS and LESS compilers. You can also use compilers (or sometimes call transpilers) to convert modern es6 or es7 javascript code into widely supported es5 javascript. The most recent updates to javascript has helped to smooth over some ugly parts (I'm looking at your Javascript OOP) and added a ton of new features. We'll get into how to use these new features in the exercises. Luckily, as long as we are just considering vanilla javascript, as opposed to coffescript or typescript, the de facto standard for compiling your code is:
```

Comment:
_Cleaned up phrasing, unpacked some of the explanations a little more_

Broadly, a compiler translates source code written in one language into another language. You might also see the word "transpiler" when you're reading about compilers, particularly in the context of JavaScript -- usually "compiling" refers to changing level of abstraction (e.g. you compile C into machine code), while "transpiling" refers to staying at about the same level of abstraction (e.g. you might transpile "new" JavaScript, ES6 or ES7, into the ES5 JavaScript you're used to). Compilers in the context of Olin.js aren't unfamiliar to you -- you've probably used SASS and LESS compilers to convert SASS/LESS to CSS.

Speaking of ES6 and ES7: the latest ECMAScript standards smooth over some currently-ugly parts of JavaScript (looking at you, object-oriented programming) and add some cool new functionalities. More on those features in the exercises!

You might also have heard of TypeScript or CoffeeScript -- they're layers of abstraction on top of vanilla JavaScript. You'd also use a transpiler to convert from one of those abstractions to straight-up JavaScript.

For now, let's just consider transpiling into ES5. In that case, usually people use: